### ISSUE TITLE
Error creating vpc and subnets

### ISSUE THREAD

**Issue by rdumi7ru** [terraform-google-modules/terraform-google-network#378]
### TL;DR

I'm trying to create a VPC with 2 subnets.
The issue is that on the first run I get a the following error:
![image](https://user-images.githubusercontent.com/25224964/192287161-e4645413-c609-40ad-86b4-7ad3ec4e7b14.png)

But the VPC network is created
On the second run it creates the subnets also, without any errors:
![image](https://user-images.githubusercontent.com/25224964/192287413-720926a3-5179-4571-a557-912dda755bb5.png)

### Expected behavior

The VPC and subnets should be created from the first run since the subnets is a required input.

### Observed behavior

The module needs to be run 2 times:
1. It creates the VPC, but when trying to create the subnets, it returns a error saying that the VPC is not found
2. It creates the subnets also.

### Terraform Configuration

```hcl
module "vpc" {
  source = "../../../modules/vpc/"

  project_id                             = var.project_id
  network_name                           = var.network_name
  routing_mode                           = var.routing_mode
  delete_default_internet_gateway_routes = true

  subnets = [
    {
      subnet_name              = "gcp-exercise-public-subnet"
      subnet_ip                = "10.10.10.0/24"
      subnet_region            = "europe-west2"
      private_ip_google_access = "true"
    },
    {
      subnet_name               = "gcp-exercise-private-subnet"
      subnet_ip                 = "10.20.10.0/24"
      subnet_region             = "europe-west1"
      private_ip_google_access  = "true"
      subnet_flow_logs          = "true"
      subnet_flow_logs_interval = "INTERVAL_10_MIN"
      subnet_flow_logs_sampling = 0.7
      subnet_flow_logs_metadata = "INCLUDE_ALL_METADATA"
    }
  ]
}
```

### Terraform Version

```sh
v1.2.8
```

**Comment by rdumi7ru** [comment]
Hello,

It was my bad, I needed to add depends_on where I declared the resources.
Now it works as intended.
